File: src/main/java/ac/kr/deu/connect/luck/foodtruck/controller/response/FoodTruckReviewResponse.java
package ac.kr.deu.connect.luck.foodtruck.controller.response;

import io.swagger.v3.oas.annotations.media.Schema;

import java.time.LocalDateTime;

@Schema(description = "리뷰 생성 요청 DTO")
public record FoodTruckReviewResponse(
        @Schema(description = "리뷰 ID", example = "1")
        Long id,
        @Schema(description = "리뷰 내용", example = "맛있어요")
        String content,
        @Schema(description = "이미지 URL", example = "https://picsum.photos/1600/900")
        String imageUrl,
        @Schema(description = "평점", example = "5")
        int rating,
        @Schema(description = "작성자 이름", example = "홍길동")
        String authorName,
        @Schema(description = "답글", example = "감사합니다.")
        String reply,
        @Schema(description = "생성일", example = "2021-08-01T00:00:00")
        LocalDateTime createdAt,
        @Schema(description = "수정일", example = "2021-08-01T00:00:00")
        LocalDateTime updatedAt
) {
}

File: src/main/kotlin/ac/kr/deu/connect/luck/foodtruck/controller/response/FoodTruckReviewResponse.kt
package ac.kr.deu.connect.luck.foodtruck.controller.response

import ac.kr.deu.connect.luck.foodtruck.entity.FoodTruckReview
import ac.kr.deu.connect.luck.user.controller.response.UserSummaryResponse
import ac.kr.deu.connect.luck.user.entity.User
import io.swagger.v3.oas.annotations.media.Schema
import java.time.Instant

@Schema(description = "푸드트럭 리뷰 응답 DTO")
data class FoodTruckReviewResponse(
    @Schema(description = "리뷰 ID", example = "1")
    val id: Long = 0L,
    @Schema(description = "리뷰 작성자 정보")
    val author: UserSummaryResponse,
    @Schema(description = "리뷰 내용", example = "맛있어요!")
    var content: String,
    @Schema(description = "평점", example = "5")
    var rating: Int,
    @Schema(description = "이미지 목록", example = "[\"image1.jpg\", \"image2.jpg\"]")
    val images: List<String>,
    @Schema(description = "답글", example = "감사합니다!")
    var reply: String? = null,
    @Schema(description = "리뷰 작성일", example = "2021-08-01T00:00:00Z")
    val createdAt: Instant = Instant.now(),
) {
    companion object {
        fun from(
            review: FoodTruckReview,
            author: User,
        ): FoodTruckReviewResponse {
            return FoodTruckReviewResponse(
                id = review.id,
                author = UserSummaryResponse.from(author),
                content = review.content,
                rating = review.rating,
                images = review.images ?: emptyList(),
                reply = review.reply,
                createdAt = review.createdAt,
            )
        }
    }
}
